packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,139 @@
+const spacingUnit = '--DataGrid-t-spacing-unit';
+
+// NOTE: Breakpoints can't come from the theme because we need access to them at
+// initialization time and media-queries can't use CSS variables. For users with
+// custom breakpoints, we might want to provide a way to configure them globally
+// instead of through the theme.
+const breakpoints = {
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+export const vars = {
+  /* Helpers */
+  /** @usage `...vars.props(vars.typography.body)` */
+  props,
+  breakpoints,
+  spacing,
+  spacingUnit,
+
+  /* Variables */
+  colors: {
+    border: {
+      base: '--DataGrid-t-colors-border-base',
+    },
+    foreground: {
+      base: '--DataGrid-t-colors-foreground-base',
+      muted: '--DataGrid-t-colors-foreground-muted',
+      accent: '--DataGrid-t-colors-foreground-accent',
+      disabled: '--DataGrid-t-colors-foreground-disabled',
+    },
+    background: {
+      base: '--DataGrid-t-colors-background-base',
+      // FIXME(kenan): validate `pinned`
+      pinned: '--DataGrid-t-colors-background-pinned',
+      // FIXME(kenan): validate `overlay` and `backdrop`
+      overlay: '--DataGrid-t-colors-background-overlay',
+      backdrop: '--DataGrid-t-colors-background-backdrop',
```

Comment:
@KenanYusuf I have also added `backdrop`, which is for the semi-transparent background of the `GridOverlay`. There is a naming conflict because I think you meant "overlay" to be for floating elements like panels or menus, but we already have `GridOverlay` using that name and needing a background. Not sure how you'd prefer to resolve the naming, let me know what you prefer.